.gitignore

```diff
@@ -16,3 +16,9 @@ exchanges/*/LICENSE
 # TODO: Figure out how to remove these:
 tmp/
 dist/
+examples/react-urql-retry/public/favicon.ico
```

Comment:
Same suggestions as per: https://github.com/FormidableLabs/urql/pull/1573#pullrequestreview-646280627
(i.e. `examples/with-retry` naming and unified `gitignore` changes)